internal/sdkprovider/service/kafka/kafka_topic_wait.go

```diff
@@ -96,6 +112,18 @@ func (w *kafkaTopicAvailabilityWaiter) refresh() error {
 		log.Printf("[DEBUG] Kafka Topic queue: %+v", queue)
 		v2Topics, err := w.Client.KafkaTopics.V2List(w.Project, w.ServiceName, queue)
 		if err != nil {
+			// V2 Kafka Topic endpoint retrieves 404 when one or more topics in the batch
+			// do not exist but does not say which ones are missing. Therefore, we need to
+			// identify the none existing topics.
+			if aiven.IsNotFound(err) {
+				// If topic is missing in V1 list then it does not exist, flagging it as missing
+				for _, t := range queue {
+					if !slices.Contains(c.GetV1List(w.Project, w.ServiceName), t) {
+						c.DeleteFromQueueAndMarkMissing(w.Project, w.ServiceName, t)
+					}
+				}
+				return nil
+			}
```

Comment:
I think if you put this right before line 113 that would safe you the API call.
Especially if you try to create multiple topics: if I get it right you'll get 404 first, and then it goes to be deleted.